packages/taiko-client/driver/preconf_blocks/api.go

```diff
@@ -357,6 +378,127 @@ func (s *PreconfBlockAPIServer) BuildPreconfBlock(c echo.Context) error {
 	return c.JSON(http.StatusOK, BuildPreconfBlockResponseBody{BlockHeader: header})
 }
 
+// BuildPreconfBlockV2 handles a preconfirmation commitment (post-transition) request.
+// It stores the txlist preimage for deterministic reconstruction, publishes the commitment
+// to the P2P network, reconstructs and inserts the block locally, and responds with the header.
+//
+//	@Summary 	    Insert a preconfirmation commitment (V2)
+//	@Description	Accepts a SignedCommitment and compressed txlist for deterministic reconstruction post-transition.
+//	@Param  		request body BuildPreconfBlockV2RequestBody true "preconfirmation commitment request body"
+//	@Accept	  	json
+//	@Produce	json
+//	@Success	200		{object} BuildPreconfBlockV2ResponseBody
+//	@Router		/preconfBlocksV2 [post]
+func (s *PreconfBlockAPIServer) BuildPreconfBlockV2(c echo.Context) error {
+	// Enable only after transition timestamp
+	if !s.preconfTransitionActive() {
+		return s.returnError(c, http.StatusBadRequest, errors.New("preconfirmation V2 API not active yet"))
+	}
+	start := time.Now()
+	defer func() {
+		elapsedMs := time.Since(start).Milliseconds()
+		metrics.DriverPreconfBuildPreconfBlockDuration.Observe(float64(elapsedMs) / 1_000)
+		log.Debug("BuildPreconfBlockV2 completed", "elapsed", fmt.Sprintf("%dms", elapsedMs))
+	}()
+
+	ctx := context.Background()
+
+	// Check if preconfirmation is enabled.
+	if s.rpc.PacayaClients.TaikoWrapper != nil {
```

Comment:
i dont think we will have a preconfRouter anymore post-Shasta and thus post-Whitelist. it is removed from the repository. i think preconfirmations dont have a way to be "disabled" and only fallback proposing will exist anymore. @ggonzalez94 correct me if im wrong?